pkg/kallsyms/kallsyms_test.go

```diff
@@ -224,3 +226,258 @@ func BenchmarkKernelSymbolAddressPreLoading(b *testing.B) {
 		require.Equal(b, true, exists)
 	}
 }
+
+func TestNewKAllSymsFromReaderErrors(t *testing.T) {
+	tests := []struct {
+		name        string
+		input       string
+		expectedErr string
+	}{
+		{
+			name:        "invalid_format_less_fields",
+			input:       "0000000000000000 A",
+			expectedErr: "line \"0000000000000000 A\" has less than 3 fields",
+		},
+		{
+			name:        "invalid_address",
+			input:       "INVALID_ADDR A fixed_percpu_data",
+			expectedErr: "strconv.ParseUint",
+		},
+		{
+			name:        "malformed_line",
+			input:       "0000000000000000",
+			expectedErr: "line \"0000000000000000\" has less than 3 fields",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			reader := strings.NewReader(tt.input)
+			_, err := NewKAllSymsFromReader(reader)
+			require.Error(t, err)
+			require.Contains(t, err.Error(), tt.expectedErr)
+		})
+	}
+}
+
+func TestLookupByInstructionPointerEdgeCases(t *testing.T) {
```

Comment:
The are some edge cases that are already tested in `TestCustomKAllSymsInstructionPointer`. Could you instead of writing a new function update that?
Especially with the helpful test names you did in this function